mypyc/transform/uninit.py

```diff
@@ -183,8 +182,7 @@ def update_register_assignments_to_set_bitmap(
                         IntOp.OR,
                         op.line,
                     )
-                    new_ops.append(new)
-                    new_ops.append(Assign(reg, new))
+                    new_ops.extend((new, Assign(reg, new)))
```

Comment:
This may be slower when compiled with mypyc.

Reply:
Building off my last comment, here are some benchmarks for `append()` vs `extend()`. Here is the relevant portions of the benchmarking code with the new snippets added:

```python
for _ in range(10_000_000):
    ...
    nums: list[int] = []

    # append
    # nums.append(1)
    # nums.append(2)

    # extend
    nums.extend((1, 2))
    ...
```

I commented out only the `append` and `extend` methods for each test, and here are the results (units are in seconds):

| Mode        | `append` | `extend` |
|-------------|------------------|-----------------------|
| Interpreted | 3.114            | 2.559                 |
| Compiled    | 7.573            | 6.874                 |

Shockingly, those snippets are much slower when compiled. I moved `nums` out of the loop, meaning only the append/extend part would be tested, and this is what I got:

| Mode        | `append` | `extend` |
|-------------|------------------|-----------------------|
| Interpreted | 2.533            | 2.019                 |
| Compiled    | 2.532            | 2.038                 |

Meaning they are pretty much on par with each other.

And, for comparison, here is the speed comparison for creating an empty list in a loop (with no append/extend):

| Mode        | `x = []` |
|-------------|----------|
| Interpreted | 1.995    |
| Compiled    | 5.396    |

Perhaps my test methodology is wrong, but we might want to look into why allocating empty lists is so much slower.

Reply:
Thanks for running the experiments! The results don't look like what I'd have expected -- very interesting. I'll investigate this.